pkg/services/ngalert/api/api_alertmanager_base.go

```diff
@@ -0,0 +1,117 @@
+/*Package api contains base API implementation of unified alerting
+ *
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ *
+ * Need to remove unused imports.
+ */
+package api
+
+import (
+	"net/http"
+
+	"github.com/go-macaron/binding"
+	apimodels "github.com/grafana/alerting-api/pkg/api"
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+type AlertmanagerApiService interface {
+	RouteCreateSilence(*models.ReqContext, apimodels.SilenceBody) response.Response
+	RouteDeleteAlertingConfig(*models.ReqContext) response.Response
+	RouteDeleteSilence(*models.ReqContext) response.Response
+	RouteGetAlertingConfig(*models.ReqContext) response.Response
+	RouteGetAmAlertGroups(*models.ReqContext) response.Response
+	RouteGetAmAlerts(*models.ReqContext) response.Response
+	RouteGetSilence(*models.ReqContext) response.Response
+	RouteGetSilences(*models.ReqContext) response.Response
+	RoutePostAlertingConfig(*models.ReqContext, apimodels.UserConfig) response.Response
+	RoutePostAmAlerts(*models.ReqContext, apimodels.PostableAlerts) response.Response
+}
+
+type AlertmanagerApiBase struct {
+	log log.Logger
+}
+
+func (api *API) RegisterAlertmanagerApiEndpoints(srv AlertmanagerApiService) {
+	api.RouteRegister.Group("", func(group routing.RouteRegister) {
+		group.Post(toMacaronPath("/alertmanager/{DatasourceId}/api/v2/silences"), binding.Bind(apimodels.SilenceBody{}), routing.Wrap(srv.RouteCreateSilence))
```

Comment:
Perhaps more of a design thing, but I'm curious about what datasourceId is specifically?

Reply:
`DatasourceId ` is used for grafana proxying the requests to cloud.
It will be something fixed (`grafana-managed`) for requests to grafana.

Reply:
This refers to the backing datasource that manages the associated payload. In the case you linked, this would either be Grafana itself or an Alertmanager.

Reply:
Not any sort of blocker, but since DatasourceId is another thing in Grafana, a different property name might be nice (but not critical). Maybe something like `Manager`?

Reply:
I would also like a different property name. The other option would be to introduce for grafana similar endpoints without this path parameter.